tectonic/xetexini.c

```diff
@@ -3736,7 +3736,7 @@ tt_run_engine(char *dump_name, char *input_file_name)
     /* Make this something invariant so that we can use XDV files to test
      * reproducibility of the engine output. */
 
-    output_comment = "tectonic";
+    output_comment = xstrdup("tectonic");
```

Comment:
This could probably be also solved by declaring `output_comment` as `const char *`.